### PR TITLE
[build] create xds_end2end_test_lib visibility tag

### PR DESCRIPTION
**Pull request by apolcyn**
As title

### CODE REVIEW

File: bazel/grpc_build_system.bzl [hunk 1]
```diff
@@ -120,6 +120,7 @@ def _update_visibility(visibility):
         "tsi": PRIVATE,
         "xds": PRIVATE,
         "xds_client_core": PRIVATE,
+        "xds_end2end_test_lib": PRIVATE,
```
Comment (markdroth) [inline bazel/grpc_build_system.bzl:123]: Please call this "xds_end2end_test_utils", to differentiate it from the "xds_end2end_test_lib" build target, which does not actually use this visibility tag. :)

File: bazel/grpc_build_system.bzl [hunk 2]
```diff
@@ -27,13 +27,13 @@
 Contains macros used throughout the repo.
 """
 
+load("@build_bazel_rules_apple//apple:ios.bzl", "ios_unit_test")
```
Comment (markdroth) [inline bazel/grpc_build_system.bzl:30]: Why move these lines?

**Review by markdroth — APPROVED** [review]

File: bazel/grpc_build_system.bzl (hunk 2 above)
Comment (apolcyn) [reply bazel/grpc_build_system.bzl]: These were from the internal formatter. I was unsure at first :) , but I've removed them to not add cruft to this PR's diff

File: bazel/grpc_build_system.bzl (hunk 1 above)
Comment (apolcyn) [reply bazel/grpc_build_system.bzl]: done

**Review by markdroth — APPROVED** [review]

**Pull request closed without merging**